recipes/fmt/all/conanfile.py

```diff
@@ -71,44 +63,48 @@ def package_id(self):
             del self.info.options.with_fmt_alias
 
     def source(self):
-        tools.get(**self.conan_data["sources"][self.version],
-                  destination=self._source_subfolder, strip_root=True)
-
-    def _configure_cmake(self):
-        if self._cmake:
-            return self._cmake
-        self._cmake = CMake(self)
-        self._cmake.definitions["FMT_DOC"] = False
-        self._cmake.definitions["FMT_TEST"] = False
-        self._cmake.definitions["FMT_INSTALL"] = True
-        self._cmake.definitions["FMT_LIB_DIR"] = "lib"
+        tools.get(**self.conan_data["sources"][self.version], destination=self.source_folder, strip_root=True)
+
+    def layout(self):
+        cmake_layout(self)
+
+    def generate(self):
+        cmake_toolchain = CMakeToolchain(self)
+
+        cmake_toolchain.variables["FMT_DOC"] = False
+        cmake_toolchain.variables["FMT_TEST"] = False
+        cmake_toolchain.variables["FMT_INSTALL"] = True
+        cmake_toolchain.variables["FMT_LIB_DIR"] = "lib"
+
         if self._has_with_os_api_option:
-            self._cmake.definitions["FMT_OS"] = self.options.with_os_api
-        self._cmake.configure(build_folder=self._build_subfolder)
-        return self._cmake
+            cmake_toolchain.variables["FMT_OS"] = self.options.with_os_api
+
+        cmake_toolchain.generate()
 
     def build(self):
         for patch in self.conan_data.get("patches", {}).get(self.version, []):
             tools.patch(**patch)
+
         if not self.options.header_only:
-            cmake = self._configure_cmake()
+            cmake = CMake(self)
+            cmake.configure()
             cmake.build()
 
     def package(self):
-        self.copy("LICENSE.rst", dst="licenses", src=self._source_subfolder)
+        self.copy("LICENSE.rst", dst="licenses")
         if self.options.header_only:
-            self.copy("*.h", dst="include", src=os.path.join(self._source_subfolder, "include"))
+            self.copy("*.h", dst="include")
         else:
-            cmake = self._configure_cmake()
+            cmake = CMake(self)
             cmake.install()
             tools.rmdir(os.path.join(self.package_folder, "lib", "cmake"))
             tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
             tools.rmdir(os.path.join(self.package_folder, "share"))
 
     def package_info(self):
-        self.cpp_info.names["cmake_find_package"] = "fmt"
-        self.cpp_info.names["cmake_find_package_multi"] = "fmt"
-        self.cpp_info.names["pkg_config"] = "fmt"
+        self.cpp_info.set_property("cmake_find_package", "fmt")
+        self.cpp_info.set_property("cmake_find_package_multi", "fmt")
+        self.cpp_info.set_property("pkg_config", "fmt")
```

Comment:
```suggestion
        self.cpp_info.set_property("cmake_target_name", "fmt")
        self.cpp_info.set_property("pkg_config_name", "fmt")
```
I think only one entry for both cmake_find_package and cmake_find_package_multi is required.